Lib/fontTools/cffLib/specializer.py

```diff
@@ -26,26 +27,62 @@ def programToString(program):
 	return ' '.join(str(x) for x in program)
 
 
-def programToCommands(program):
+def programToCommands(program, deltasToSourceValues=None, numRegions=0):
 	"""Takes a T2CharString program list and returns list of commands.
 	Each command is a two-tuple of commandname,arg-list.  The commandname might
 	be empty string if no commandname shall be emitted (used for glyph width,
 	hintmask/cntrmask argument, as well as stray arguments at the end of the
-	program (¯\_(ツ)_/¯)."""
+	program. The latter happens in subroutines.
+	Note that each 'blend' operator and its argument list is  replaced by a
+	list of arguments in which each argument item is a list of source font
+	values.
+	'deltasToSourceValues' can be from varLib.varStore.VarDataItem."""
 
 	width = None
 	commands = []
 	stack = []
+	seen_blend = False
+	seen_op = False
+	num_sources = numRegions + 1
+
 	it = iter(program)
 	for token in it:
 		if not isinstance(token, basestring):
 			stack.append(token)
 			continue
-
-		if width is None and token in {'hstem', 'hstemhm', 'vstem', 'vstemhm',
-					       'cntrmask', 'hintmask',
-					       'hmoveto', 'vmoveto', 'rmoveto',
-					       'endchar'}:
+		if token == 'blend':
+			# if we see a blend, it is a CFF charstring, and there
```

Comment:
did you mean perhaps this
```suggestion
			# if we see a blend, it is a CFF2 charstring, and there
```